build.cmd

```diff
@@ -16,24 +16,20 @@ IF NOT EXIST %BUILD_FOLDER% (
     md %BUILD_FOLDER%
 )
 
-IF NOT EXIST %NUGET_PATH% (
-    IF NOT EXIST %CACHED_NUGET% (
-        echo Downloading latest version of NuGet.exe...
-        IF NOT EXIST %LocalAppData%\NuGet ( 
-            md %LocalAppData%\NuGet
-        )
-        @powershell -NoProfile -ExecutionPolicy unrestricted -Command "$ProgressPreference = 'SilentlyContinue'; Invoke-WebRequest 'https://dist.nuget.org/win-x86-commandline/%NUGET_VERSION%/nuget.exe' -OutFile '%CACHED_NUGET%'"
-    )
-
-    copy %CACHED_NUGET% %NUGET_PATH% > nul
+IF EXIST %NUGET_PATH% goto installkorebuild
+IF NOT EXIST %CACHED_NUGET% (
+    echo Downloading latest version of NuGet.exe...
+    IF NOT EXIST %LocalAppData%\NuGet md %LocalAppData%\NuGet
```

Comment:
@victorhurdugaci nested blocks should be fine as long as nothing within an `IF` block depends on a variable set earlier in the same block. For that case, labels may be harder to read but are more understandable than CMD's delayed expansion feature.